### PR TITLE
Lifted import_vcf

### PR DESCRIPTION
**Pull request by tpoterba**

### CODE REVIEW

File: python/hail/methods/impex.py [hunk 1]
```diff
@@ -629,8 +630,124 @@ def get_vcf_metadata(path):
...
+    ".vcf.bgz" accordingly. If you actually have a standard gzipped file, it is
+    possible to import it to Hail using the ``force`` optional parameter.
+    However, this is not recommended -- all parsing will have to take place on
+    one node because gzip decompression is not parallelizable. In this case,
+    import will take significantly longer.
+
+
+    :func:`.import_vcf` does not perform deduplication - if the provided VCF(s)
+    contain multiple records with the same chrom, pos, ref, alt, all these
+    records will be imported as-is and will not be collapsed into a single
```
Comment (cseed) [inline python/hail/methods/impex.py]: > imported as-is

clear if this means multiple rows.

File: python/hail/methods/impex.py [hunk 2]
```diff
@@ -629,8 +630,124 @@ def get_vcf_metadata(path):
...
+        Using the **FILTER** field:
+
+        The information in the FILTER field of a VCF is contained in the
+        ``filters`` row field. This annotation is a ``Set[String]`` and can be
+        queried for filter membership with expressions like
+        ``ds.filters.contains("VQSRTranche99.5...")``. Variants that are flagged
+        as "PASS" will have no filters applied; for these variants,
+        ``ds.filters.is_empty()`` is ``True``. Thus, filtering to PASS variants
+        can be done with :meth:`.MatrixTable.filter_rows` as follows:
+
```
Comment (cseed) [inline python/hail/methods/impex.py]: extra newline

File: python/hail/methods/impex.py [hunk 3]
```diff
@@ -629,8 +630,124 @@ def get_vcf_metadata(path):
...
+
+        >>> pass_ds = dataset.filter_rows(ds.filters.is_empty())
+
```
Comment (cseed) [inline python/hail/methods/impex.py]: extra newline

File: python/hail/methods/impex.py [hunk 4]
```diff
@@ -629,8 +630,124 @@ def get_vcf_metadata(path):
...
+    Ensure that the VCF file is correctly prepared for import: VCFs should
+    either be uncompressed (*.vcf*) or block compressed (*.vcf.bgz*). If you
+    have a large compressed VCF that ends in *.vcf.gz*, it is likely that the
+    file is actually block-compressed, and you should rename the file to
+    ".vcf.bgz" accordingly. If you actually have a standard gzipped file, it is
+    possible to import it to Hail using the ``force`` optional parameter.
+    However, this is not recommended -- all parsing will have to take place on
+    one node because gzip decompression is not parallelizable. In this case,
+    import will take significantly longer.
+
```
[the remark this answers is not in the document]
Comment (cseed) [reply python/hail/methods/impex.py]: Do you need a code-block directive here?

File: python/hail/methods/impex.py [hunk 5]
```diff
@@ -629,8 +630,124 @@ def get_vcf_metadata(path):
...
+        ``ds.filters.contains("VQSRTranche99.5...")``. Variants that are flagged
+        as "PASS" will have no filters applied; for these variants,
+        ``ds.filters.is_empty()`` is ``True``. Thus, filtering to PASS variants
+        can be done with :meth:`.MatrixTable.filter_rows` as follows:
+
+
+        >>> pass_ds = dataset.filter_rows(ds.filters.is_empty())
+
+
+    **Column Fields**
```
Comment (cseed) [inline python/hail/methods/impex.py]: Should we use

```
Column Fields
-------------------
```

here?

File: python/hail/methods/impex.py [hunk 6]
```diff
@@ -629,8 +630,124 @@ def get_vcf_metadata(path):
...
+    the same header and the same set of samples in the same order (e.g., a
+    dataset split by chromosome). Files can be specified as :ref:`Hadoop glob
+    patterns <sec-hadoop-glob>`.
+
+
+    Ensure that the VCF file is correctly prepared for import: VCFs should
+    either be uncompressed (*.vcf*) or block compressed (*.vcf.bgz*). If you
+    have a large compressed VCF that ends in *.vcf.gz*, it is likely that the
+    file is actually block-compressed, and you should rename the file to
+    ".vcf.bgz" accordingly. If you actually have a standard gzipped file, it is
```
Comment (cseed) [inline python/hail/methods/impex.py]: You use quotes here but double-star for extensions above.

File: python/hail/methods/impex.py [hunk 7]
```diff
@@ -629,8 +630,124 @@ def get_vcf_metadata(path):
...
+
+    **Row Fields**
+
+    - **v** (*Variant*) -- Row key. This is the variant created from the CHROM,
+      POS, REF, and ALT fields.
+    - **filters** (*Set[String]*) -- Set containing all filters applied to a
+      variant.
+    - **rsid** (*String*) -- rsID of the variant.
+    - **qual** (*Double*) -- Floating-point number in the QUAL field.
+    - **info** (*Struct*) -- All INFO fields defined in the VCF header
```
Comment (cseed) [inline python/hail/methods/impex.py]: You use double backticks for types above but stars here.  I guess we should be consistent?

File: python/hail/methods/impex.py [hunk 8]
```diff
@@ -629,8 +630,124 @@ def get_vcf_metadata(path):
...
+    in the VCF header. The types of these fields are generated according to the
+    same rules as INFO fields, with one difference -- "GT" and other fields
+    specified in `call_fields` will be read as :class:`.TCall`.
+
+    Parameters
+    ----------
+    path : :obj:`str` or :obj:`list` of :obj:`str`
+        VCF file(s) to read.
+    force : :obj:`bool`
+        If ``True``, load .gz files serially. No downstream operations can be
```
Comment (cseed) [inline python/hail/methods/impex.py]: extension, double star?

Comment (cseed) [reply python/hail/methods/impex.py]: a few more times below.

File: python/hail/methods/impex.py [hunk 9]
```diff
@@ -629,8 +630,124 @@ def get_vcf_metadata(path):
...
+    -------
+    :class:`.MatrixTable`
+    """
+    rg = reference_genome if reference_genome else default_reference()
+
+    if contig_recoding:
+        contig_recoding = TDict(TString(), TString())._convert_to_j(contig_recoding)
+
+    jmt = Env.hc()._jhc.importVCFs(jindexed_seq_args(path), force, force_bgz, joption(header_file),
+                                joption(min_partitions), drop_samples, jset_args(call_fields), rg._jrep,
```
Comment (cseed) [inline python/hail/methods/impex.py]: identation

File: python/hail/methods/impex.py (hunk 5 above)
Comment (tpoterba) [reply python/hail/methods/impex.py]: will error out, napoleon only accepts known headers

Comment (cseed) [reply python/hail/methods/impex.py]: Ah, good to know!  I've been tempted before.

File: python/hail/methods/impex.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (tpoterba) [reply python/hail/methods/impex.py]: why?

File: python/hail/methods/impex.py (hunk 7 above)
Comment (tpoterba) [reply python/hail/methods/impex.py]: We don't really follow a convention elsewhere, I guess here's an opportunity to set it. I think we should use italics (single backticks) for field names, and link to the Type class. I'l change that here.

Comment (tpoterba) [reply python/hail/methods/impex.py]: also, oops -- left "Double"!

File: python/hail/methods/impex.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (cseed) [reply python/hail/methods/impex.py]: nvm

**Pull request merged**